packages/common/src/models/Track.ts

```diff
@@ -89,6 +91,41 @@ export type PremiumConditions = {
   usdc_purchase?: PremiumConditionsUSDCPurchase
 }
 
+// Must include undefined because during upload flow, the user could have selected
+// Collectible-gated option without specifying a collection yet. But afterweards
+// when we read PremiumConditions, nft_collection should always exist.
+type PremiumConditionsCollectibleGated = PremiumConditionsBase & {
```

Comment:
Hmm... feels sort of wrong to have the discrimination unions have optional types from the other cases. One could have a `USDC_PURCHASE` type with a possible defined `nft_collection` field, which half-way defeats the purpose of this discriminated union imo. Down to brainstorm a way to avoid this

Reply:
This is only a migratory step - ideally we convert all consumers to check against `type` instead of existence of `nft_collection`, `follower_user_id` etc and then can come back and delete `PremiumConditionsBase`.
Or are you thinking there's a way to avoid this even in the intermediary state?